src/games/strategy/engine/framework/GameRunner2.java

```diff
@@ -172,14 +186,25 @@ public void run() {
         }
       });
     } catch (final Exception e) {
-      // just don't show the wait window
+      ClientLogger.logQuietly(e);
+      if(s_waitWindow != null ) {
+        s_waitWindow.dispose();
+      }
     }
     setupProxies();
     showMainFrame();
-    // lastly, check and see if there are new versions of TripleA out
-    checkForUpdates();
+    checkForGameEngineUpdates();
   }
 
+  /**
+   * Used to set the object that will handle uncaught exceptions, generally these are exceptions
+   * thrown from the Swing and other threads. Generally useful to help test.
+   */
+  public static void setErrorHandler(UncaughtExceptionHandler handler) {
```

Comment:
It's here so a test can inject the error handler. The test would check the error handler after each test run to see if exceptions were thrown. That test is not yet checked in:
- right now i'm focusing on moving over as much of the self contained work that I have onto the updated re-formatted master branch, so this PR is really not quite as clean/nice as I would like. I appreciate you bearing with me as much as you have though.